internal/stream_metadata.go

```diff
@@ -0,0 +1,60 @@
+package internal
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+
+	"github.com/wal-g/tracelog"
+)
+
+const (
+	SplitMergeStreamBackup   = "SPLIT_MERGE_STREAM_BACKUP"
+	SingleStreamStreamBackup = "STREAM_BACKUP"
+)
+
+type BackupStreamMetadata struct {
+	Type        string `json:"type"`
```

Comment:
If this metadata does not consume much space, maybe just add it to the sentinel?

Reply:
it will increase coupling between uploader metadata and sentinel